File: NextTaskClient/src/shared/types/comment.ts
// Типы комментариев. Централизовано для всего фронтенда
// Комментарии в коде — на русском, согласно правилам проекта

export interface CommentAuthor {
	id: number;
	name?: string;
	email: string;
	avatar?: string;
}

export interface CommentItem {
	id: number;
	content: string;
	author_id: number; // для совместимости с прямым ответом бэкенда
	author?: CommentAuthor; // может прийти в расширенном ответе
	task_id: number;
	created_at?: string; // ISO
}

export interface CreateCommentRequest {
	content: string;
}

export interface CommentsQuery {
	limit?: number; // 1..100, по умолчанию 50
	offset?: number; // >=0, по умолчанию 0
	order?: "asc" | "desc"; // порядок сортировки на бэкенде
}

File: NextTaskServer/app/schemas/comment.py
from pydantic import BaseModel, Field
from typing import Optional, List
from datetime import datetime

class CommentAuthor(BaseModel):
    id: int
    name: Optional[str] = None
    email: str
    avatar: Optional[str] = None

class CommentBase(BaseModel):
    content: str

class CommentCreate(CommentBase):
    pass

class CommentUpdate(BaseModel):
    content: str

class CommentResponse(CommentBase):
    id: int
    author_id: int
    author: Optional[CommentAuthor] = None
    task_id: int
    created_at: Optional[datetime] = None

class CommentsResponse(BaseModel):
    comments: List[CommentResponse]

class CommentsCountResponse(BaseModel):
    count: int

class CommentsQuery(BaseModel):
    limit: Optional[int] = 50
    offset: Optional[int] = 0
    order: Optional[str] = "desc"
